File: backend_spring_boot/bitmind/src/main/java/com/backend/bitmind/Model/Anuncio.java
package com.backend.bitmind.Model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "anuncios")
public class Anuncio {

    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    @Column(name = "id_anuncio")
    private int idAnuncio;

    @Column(name = "titulo")
    private String titulo;

    @Lob
    @Column(name = "descripcion", columnDefinition = "TEXT")
    private String descripcion;

    @Column(name = "imagen")
    private String imagen;

    @Column(name = "fecha_creacion")
    private LocalDateTime fechaCreacion;

    @ManyToOne
    @JoinColumn(name = "id_usuario", referencedColumnName = "id_usuario")
    private Usuario usuario;
}

File: frontend_kotlin_user/app/src/main/java/com/nunez/abraham/bitmind_frontend_movil/models/Anuncio.kt
package com.nunez.abraham.bitmind_frontend_movil.models

import java.time.LocalDateTime

data class Anuncio(
    val idAnuncio: Int,
    val titulo: String,
    val descripcion: String,
    val imagen: String,
    val fechaCreacion: LocalDateTime,
    val usuario: Usuario
)
